### DISCUSSION TITLE
Expand OSC alpha/transparency with Snap Cursor metronome

### DISCUSSION THREAD

**Discussion by gmschroeder** [brunoherbelin/vimix#145] [General]
Changing the Snap Cursor alpha values via OSC would make my life much, much simpler and give Vimix a lot more artistic flexibility. 
For example, I'm literally rebuilding the Snap Cursor's "Metronome" setting in Pure Data with information from Ableton Link and /vimix/TARGET/loom or /vimix/TARGET/alpha's optional arguments. And math. Then I realized that's what I was doing. I just want to do it from an OSC source instead of clicking and dragging in the Mixing page.
Pretty please let me use snap cursor from outside Vimix.
Something like /vimix/TARGET/snap/metro <target alpha value> and /vimix/TARGET/snap/grid <target alpha value>  on or off with > 0.5 and < 0.5. Off by default. I don't know how much I'd use the unique settings for each one from outside Vimix, though. Maybe leave the slider on each setting just in the GUI?
What do y'all think?

**Comment by brunoherbelin** [comment]
Thanks for the suggestion and glad that you find the snap cursors useful!

Yes, all that is very doable, and i will gladly implement it in the coming weeks (vacation time...): OSC targets you describe are relatively clear already, but please do not hesitate to give a detailed list of all parameters we could change so that I would do the implementation once for all!

**Reply by gmschroeder** [reply]
1) Ways to navigate and 'replace' sources via OSC would aid how I work. In other software (and hardware) I load up 3-4 sources at a time then replace them one by one as I need to. I don't know if that's how people envisioned Vimix working. I feel like Vimix is less "jam"-y, and that's part of it.

That workflow would need a predictable, internal list directory. (i.e. "Not Recent Files") Having Sources in alphanumerical order in that list directory when I'm picking a new source would make a lot more sense.

2) Why can't we replace a Source that has a Clone?

3) Is there a backstory to why Vimix doesn't have midi at all? Will it ever? I'm begging for OSC receives from the developer on 5-year-old software so I can make a separate piece of software to translate midi to OSC. I feel a little silly doing it.

I know this isn't commercial software, but I did what Vimix is supposed to do and then some (play back videos, route live video, run shaders as generators and effects, control all that with common hardware, start fiddling with audioreactivity, and learn how to do it with skimming the manual) in a weekend with a free demo of $80 software, and MIDI is 90% of why.
I'd like a better sense of what's in store for Vimix before I put more work into using it. That was extremely frustrating.
